adaptive_scheduler/configdb_connections.py

```diff
@@ -88,7 +89,18 @@ def get_all_active_instruments(self):
         json_results = r.json()
         if 'results' not in json_results:
             raise ConfigDBError("get_all_active_instruments failed: ConfigDB returned no results")
-        return json_results['results']
+        # Filter out instruments in non-included telescope_classes if they are specified
+        if self.telescope_classes:
+            # non-empty telescope_classes means we have something to filter by
+            instruments = []
+            for instrument in json_results['results']:
+                split_string = instrument['__str__'].lower().split('.')
+                telescope_class = split_string[2][:3]
+                if telescope_class.lower() in self.telescope_classes.lower():
```

Comment:
When I first started reading through this I assumed that `telescope_classes` was a list, until I got here and saw it being `.lower()`-ed and realized it was a comma delimited string. I wonder if it would make more sense to handle telescope_classes as a list for clarity.

Reply:
fixed - stored and passed around as list internally